### ISSUE TITLE
Ability to limit supported TLS versions and disable certain versions

### ISSUE THREAD

**Issue by sgringwe** [heptio/contour#185]
PCI compliance states that TLS 1.0 cannot be enabled starting June 2018 (https://blog.pcisecuritystandards.org/are-you-ready-for-30-june-2018-sayin-goodbye-to-ssl-early-tls).

It'd be great to have a way to disable TLS 1.0 in Contour or generally configure supported versions.

**Comment by davecheney** [comment]
This is a _very_ good idea. Currently the `tls_params` field in the listener's TLS context is not specified so I'm guessing it defaults to whatever the default of the ssl library that Envoy links against.
```
https://www.envoyproxy.io/docs/envoy/latest/api-v2/sds.proto.html#envoy-api-msg-commontlscontext
```
 This isn't great (and it isn't great to guess).

What I think we should do is twofold

- [ ] send a PR to provide a reasonably modern `tls_params` set of parameters. The values of reasonable and modern can be negotiated later. My view is the cipher suite params should tend towards the restrictive.
- [ ] iff we get bug reports that people can't use insecure versions of browsers and appliances to communicate with their applications, we can add the ability to specify the params suite as an annotation to the ingress object.

**Comment by sevein** [comment]
The default [ciphers](https://github.com/envoyproxy/envoy/blob/v1.5.0/source/common/ssl/context_config_impl.cc#L15-L33) and [curves](https://github.com/envoyproxy/envoy/blob/9ed62923a8ff6745407046c4451ce757348d966f/source/common/ssl/context_config_impl.cc#L35) in Envoy seem pretty sane to me.

Regarding to the protocols supported, the following table shows the current defaults and the change that I would like to suggest:

| Property                                  | Envoy's interpretation of zero value                                                                        | Suggested change                                                                                     |  
| ----------------------------------------- | ----------------------------------------------------------------------------------------------------------- | -----------------------                                                                              |  
| `TlsParameters.TlsMinimumProtocolVersion` | [TLS 1.0](https://github.com/envoyproxy/envoy/blob/v1.5.0/source/common/ssl/context_config_impl.cc#L56-L57) | [TLS 1.1](https://github.com/envoyproxy/data-plane-api/blob/master/envoy/api/v2/auth/cert.proto#L26) |  
| `TlsParameters.TlsMaximumProtocolVersion` | [TLS 1.2](https://github.com/envoyproxy/envoy/blob/v1.5.0/source/common/ssl/context_config_impl.cc#L58-L59) | None                                                                                                 |  

Support for annotations in the ingress object would be great in the long term!

**Comment by davecheney** [comment]
Some work on this was done in 0.4 by @sevein. We might do more in 0.5.

**Comment by davecheney** [comment]
Nothing to do for 0.5, bumping to 0.6

**Comment by rosskukulinski** [comment]
I believe this was resolved in #324 for Ingress via annotation, and then #596 for IngressRoute.  Please re-open if that is not the case.
